### DISCUSSION TITLE
How to use Dialogic Set value and set definition for the timeline?

### DISCUSSION THREAD

**Discussion by PuteraSinuraya** [dialogic-godot/dialogic#999] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/60869659/178404425-f058b00b-f2ca-4308-8a03-00f18f787d4a.png)

So I have noticed in the timeline when you add a text with the associated character, there is an option set portrait. When I clicked the the drop down menu, I noticed X[value], which will display a definition drop down next to it, if selected. (as seen from the screenshot).

How do I take advantage of this feature?

**Comment by Jowan-Spooner** [comment, marked as answer]
Hi. You can use this feature 
- by creating a new Value in dialogic (e.g. named 'SidekickPortrait')
- setting the Value (with the Set value event) to the name of one of your portraits (e.g. happy)
   - alternatively you can set the Value via code with `Dialogic.set_variable(name, value)` (e.g. `Dialogic.set_variable('SidekickPortrait', "happy")`)
- selecting the Value in the dropdown

This should show the characters happy portrait.

The advantage is, that you can control the characters portrait more dynamically, but (imo the workflow is honestly a bit tedious).
Hope this helps.

**Reply by PuteraSinuraya** [reply]
![image](https://user-images.githubusercontent.com/60869659/179389907-d52eb81c-851b-482f-91b9-bd4e28182810.png)
By the way. I just realized that you can make duplicate variables in the dialogic definition. If I use 
`Dialogic.Set_variable("name",value)`
would it just overwrite one of the duplicate variables or both?

**Reply by Jowan-Spooner** [reply]
Just one. If you want a specific one you can specify the whole path i believe. Like set_variable('Face/Lisa', 'happy' ).

**Comment by Yotopihi-Dev** [comment]
this could be useful if this timeline just came from a certain situation, so if the character is now scared, happy, or mad, you can let it check for what value it had gotten set before to make that possible
